### PR TITLE
Added support for running ruff

### PR DESCRIPTION
**Pull request by andy-maier**
The Python "ruff" package is used, and is run on Python >=3.7.
For details, see the commit message and issue #3144.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -362,6 +362,7 @@ help:
 	@echo "  build      - Build the source and wheel distribution archives in: $(dist_dir)"
 	@echo "  builddoc   - Build documentation in: $(doc_build_dir)"
 	@echo "  check      - Run Flake8 on sources"
+	@echo "  ruff       - Run Ruff on sources"
```
Comment (KSchopmeyer) [inline Makefile:365]: Add "an alternate line tool"

**Review by KSchopmeyer — APPROVED** [review]
Approved with one  minor comment comment

File: Makefile (hunk 1 above)
Comment (KSchopmeyer) [reply Makefile:365]: Approved with this one change

Comment (andy-maier) [reply Makefile:365]: DONE

**Pull request merged**